### PR TITLE
Implement 'all content in category' page

### PR DESCRIPTION
**Pull request by jacksonj04**
Provide initial stub pages for viewing all categories, and all content within a category.

### CODE REVIEW

File: cms/categories/templates/categories/category_detail.html [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+<div class="nhsuk-width-container">
+    <div class="nhsuk-grid-row">
+        <div class="nhsuk-grid-column-full">
+            <h1>{{ object.name }}</h1>
+        </div>
+    </div>
+
+    <div class="nhsuk-grid-row">
+        <div class="nhsuk-grid-column-two-thirds">
```
Comment (dragon-dxw) [inline cms/categories/templates/categories/category_detail.html:16]: Is there a reason to not be the full width of the page (i.e. footer width?)

File: cms/categories/templates/categories/category_detail.html [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+          <p>{{ object.description }}</p>
+          {% if category_pages %}
+            <table role="table" class="nhsuk-table-responsive">
+              <thead class="nhsuk-table__head">
+                <tr>
+                  <th>Title</th>
+                  <th>Content type</th>
+                  <th>Date published/updated</th>
```
Comment (dragon-dxw) [inline cms/categories/templates/categories/category_detail.html:24]: This feels unnecessarily ambiguous; would "Last updated" work better?

File: cms/categories/templates/categories/category_detail.html [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+              </thead>
+              <tbody class="nhsuk-table__body">
+              {% for page in category_pages %}
+                <tr class="nhsuk-table__row">
+                    <td class="nhsuk-table__cell">
+                      <a href="{{page.record.url}}">
+                      {{page.record.title}}</a>
+                    </td>
+                    <td class="nhsuk-table__cell"><span class="nhsuk-tag nhsuk-tag--blue">{{ page.tag }}</span></td>
+                    <td class="nhsuk-table__cell"><time class="date" datetime="">{{ page.record.last_published_at|date:'d M Y g:i a' }}</time></td>
```
Comment (dragon-dxw) [inline cms/categories/templates/categories/category_detail.html:35]: There's probably work to be done on confirming that this (and not latest_revision_created_at) but I'm not convinced on the right answer myself. (comment, not request for change)

File: cms/categories/urls.py [hunk 4]
```diff
@@ -0,0 +1,8 @@
+from django.urls import path
+
+from cms.categories import views
+
+urlpatterns = [
+    path("", views.CategoryListView.as_view()),
```
Comment (dragon-dxw) [inline cms/categories/urls.py:6]: This doesn't seem to work: I get redirected from categories to categories/ which matches against the line below with NoReverseMatch at /categories/
Reverse for 'category-detail' with arguments '('',)' not found. 1 pattern(s) tried: ['categories/(?P<slug>[-a-zA-Z0-9_]+)$']

File: cms/categories/views.py [hunk 5]
```diff
@@ -1,3 +1,38 @@
...
+    model = Category
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        pages = CategoryPage.objects.live()
+        category = context["object"]
+        pages = pages.filter(categorypage_category_relationship__category=category)
+        ordered_pages = pages.order_by("-last_published_at")
+        page_data = [
```
Comment (dragon-dxw) [inline cms/categories/views.py:26]: I'm happy with this duplication. If we find ourselves doing this again, we might want to consider refactoring.
(comment only)

**Review by dragon-dxw — CHANGES_REQUESTED** [review]
categories/missing-category 404s, which is perfectly fine.

The only showstopper is the /categories path handling -- the formatting width and column title are acceptable if you think there's a good reason for them.

File: cms/categories/templates/categories/category_detail.html (hunk 2 above)
Comment (jacksonj04) [reply cms/categories/templates/categories/category_detail.html:24]: This is the wording from NHS. If we get a later change request it's an easy fix.

**Review by dragon-dxw — APPROVED** [review]

**Pull request merged**